internal/driver/fetch.go

```diff
@@ -580,17 +581,35 @@ func adjustURL(source string, duration, timeout time.Duration) (string, time.Dur
 
 // httpGet is a wrapper around http.Get; it is defined as a variable
 // so it can be redefined during for testing.
-var httpGet = func(source string, timeout time.Duration) (*http.Response, error) {
+var httpGet = func(s *source, source string, timeout time.Duration) (*http.Response, error) {
```

Comment:
We shouldn't need to pass all of "s *source" here, we only need the TLS params?

Reply:
Just seemed cleaner to not create anything extra, just reuse the existing struct. If we make the `TLSParams` in `plugin.go` that should work for `Symbolizer()` as well. Isn't having the same `HTTPS*` fields in the `source` struct redundant?

Reply:
I would TLSParams in the source structure as well to avoid the duplicated fields, but (as I said) would pass the source / URL string and the TLS params alone instead of passing the whole input structure. Passing only what's needed is better than being able to avoid another little struct imo.